internal/api/middleware.go

```diff
@@ -84,3 +89,76 @@ func loggingMiddleware(ctx *middlewareContext) middleware {
 		})
 	}
 }
+
+// authMiddleware intercepts requests that need authorization to check if the user's token exists and is
+// permitted to use the endpoint
+func authMiddleware(ctx *middlewareContext) middleware {
+	AdminOnlyPaths := []struct{ method, path string }{
+		{"POST", `accounts`},
+		{"GET", `accounts`},
+		{"GET", `accounts\/\d+$`},
+		{"DELETE", `accounts\/\d+$`},
+		{"POST", `accounts\/\d+\/change_password$`},
+	}
+
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !strings.HasPrefix(r.URL.Path, "/api/v1/") {
+				next.ServeHTTP(w, r)
+				return
+			}
+			if r.Method == "POST" && strings.HasSuffix(r.URL.Path, "accounts") && !ctx.firstAccountIssued {
+				ctx.firstAccountIssued = true
```

Comment:
Providing an invalid password (for example too short) on the first attempt to create the admin user will result in this being true without a user created and there is no way to recover.

Reply:
great catch, fixed.